numpy/core/numerictypes.py

```diff
@@ -297,64 +297,78 @@ def _add_types():
             allTypes[name] = info
 _add_types()
 
+# Integers are handled so that the int32 and int64 types should agree
+# exactly with NPY_INT32, NPY_INT64. We need to enforce the same checking
+# as is done in arrayobject.h where the order of getting a bit-width match
+# is long, longlong, int, short, char.
+_integer_ctypes = ['LONG', 'LONGLONG', 'INT', 'SHORT', 'BYTE']
```

Comment:
Moved from below so that we can drop the fragile `base[-3:] == 'int' or char[0] in 'ui'` check